pwiz_tools/Shared/PanoramaClient/FilePicker.cs

```diff
@@ -0,0 +1,897 @@
+                $@"{server}{folderPath}/query-selectRows.view?schemaName=targetedms&query.queryName={queryName}&query.containerFilterName={folderFilter}";
+            if (columns != null)
+            {
+                query = $@"{query}&query.columns=";
+                var allCols = columns.Aggregate(string.Empty, (current, col) => $@"{col},{current}");
+
+                query = $@"{query}{allCols}";
+            }
+
+            if (!string.IsNullOrEmpty(sortParam))
+            {
+                query = $@"{query}&query.sort={sortParam}";
+            }
+
+            if (!string.IsNullOrEmpty(equalityParam))
+            {
+                query = $@"{query}&query.{equalityParam}~eq=";
+            }
+            return query;
+        }
+
+        /// <summary>
+        /// Takes in a query string and returns the associated JSON
+        /// </summary>
+        /// <param name="query"></param>
+        /// <returns></returns>
+        private JToken GetJson(string query)
+        {
+            var queryUri = new Uri(query);
+            var webClient = new WebClientWithCredentials(queryUri, ActiveServer.Username, ActiveServer.Password);
+            JToken json = webClient.Get(queryUri);
+            return json;
+        }
+
+
+        /// <summary>
+        /// Adds all files in a particular folder 
+        /// </summary>
+        /// <param name="newUri"></param>
+        public void AddChildFiles(Uri newUri)
+        {
+            var json = GetJson(newUri.ToString());
+            if ((int)json[@"fileCount"] != 0)
+            {
+                var files = json[@"files"];
+                foreach (dynamic file in files)
+                {
+                    var listItem = new string[5];
+                    var fileName = (string)file[@"text"];
+                    listItem[0] = fileName;
+                    var isFile = (bool)file[@"leaf"];
+                    if (isFile)
+                    {
+                        var canRead = (bool)file[@"canRead"];
+                        if (!canRead)
+                        {
+                            continue;
+                        }
+                        var size = (long)file[@"size"];
+                        var sizeObj = new FileSize(size);
+                        listItem[1] = sizeObj.ToString();
+                        listItem[4] = (string)file[@"creationdate"];
+                        var fileNode = fileName.EndsWith(EXT) ? new ListViewItem(listItem, 1) : new ListViewItem(listItem, 0);
+                        fileNode.Tag = (string)file[@"id"];
+                        fileNode.Name = (string)file[@"href"];
+                        listView.Items.Add(fileNode);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Displays the latest versions of all files in a particular folder
+        /// </summary>
+        /// <param name="path"></param>
+        private void GetLatestVersion(string path)
```

Comment:
Do you need another query here? I think you should have all the information you need to display the latest versions from the `AddQueryFies` method since you are including all the required columns in that query.

Reply:
I decided the keep the _peptideInfoQuery a class variable and just reference that query from this method so I only end up making this query one time instead of re-querying.